desktop/ui/src/main/java/org/datacleaner/user/DataCleanerHome.java

```diff
@@ -80,16 +80,38 @@ private static FileObject findDataCleanerHome() throws FileSystemException {
 
         FileObject candidate = null;
 
-        final String env = System.getenv("DATACLEANER_HOME");
+        String env = System.getenv("DATACLEANER_HOME");
         if (!StringUtils.isNullOrEmpty(env)) {
+            if (env.startsWith("~")) {
+                String userHomePath = System.getProperty("user.home");
+                env = env.replace("~", userHomePath);
+            }
             candidate = manager.resolveFile(env);
             logger.info("Resolved env. variable DATACLEANER_HOME ({}) to: {}", env, candidate);
+        } else {
+            String sysProp = System.getProperty("DATACLEANER_HOME");
+            if (!StringUtils.isNullOrEmpty(sysProp)) {
+                if (sysProp.startsWith("~")) {
+                    String userHomePath = System.getProperty("user.home");
+                    sysProp = sysProp.replace("~", userHomePath);
+                }
+                candidate = manager.resolveFile(sysProp);
+                logger.info("Resolved system property DATACLEANER_HOME ({}) to: {}", sysProp, candidate);
+            }
         }
 
         if (isUsable(candidate)) {
+            // Found a directory with conf.xml already there
             return candidate;
+        } else {
+            return initializeDataCleanerHome(candidate);
         }
```

Comment:
Very minor, but if you moved the isUsable() check into initializeDataCleanerHome (which would just return candidate if it was already initialized), you'd save a bit of code.